gateway/ln-gateway/src/rpc/mod.rs

```diff
@@ -211,7 +210,83 @@ pub fn serde_hex_serialize<T: bitcoin::consensus::Encodable, S: Serializer>(
     }
 }
 
+// MIGRATION:
+//
+// Borrow and re-implement gateway channel messaging types in parallel to
+// the legacy implementations above which were originally designed
+// for gateway REST webserver communication
```

Comment:
I guess this commit is mostly for compatibility with the current impl? Otherwise it feels like slightly overcomplicating things. I think it would be good to have some protocol diagrams that show how communication happens between gw and extension.